### PR TITLE
Undo #19, fixed upstream

### PR DESCRIPTION
**Pull request by neykov**
RetryOnRenew fixed in jclouds upstream:
  * https://issues.apache.org/jira/browse/JCLOUDS-178
  * https://issues.apache.org/jira/browse/JCLOUDS-589

Unit tests are passing, works fine with SL objectstore.
How do I launch the tests in `BlobStoreExpiryTest`? I get `Caused by: java.net.UnknownHostException: keystone-endpoint` at BlobStoreExpiryTest.java:153.

cc @andreaturli @ahgittin

### CODE REVIEW

**Comment by ahgittin** [comment]
@neykov git history gives some clues to the mysterious `keystone-endpoint`:

```
-    private void injectShortLivedTokenForSoftlayerAmsterdam() {
-        HttpToolResponse tokenHttpResponse1 = requestTokenWithExplicitLifetime("https://ams01.objectstorage.softlayer.net/auth/v1.0/v1.0", "ams01.objectstorage.softlayer.net", 
+    private void injectShortLivedTokenForKeystoneV1_1() {
+        HttpToolResponse tokenHttpResponse1 = requestTokenWithExplicitLifetime("https://keystone-endpoint/v1.1", "CHANGE_ME",
```

nicest i think would be if it is plucked from config somehow

**Comment by neykov** [comment]
Fixed merge conflicts. Any comments? (ready for merge).

File: locations/jclouds/src/test/java/brooklyn/entity/rebind/persister/jclouds/BlobStoreExpiryTest.java [hunk 1]
```diff
@@ -91,7 +91,7 @@
     private String provider;
     private String endpoint;
 
-    public synchronized BlobStoreContext getBlobStoreContext(boolean applyFix) {
+    public synchronized BlobStoreContext getBlobStoreContext() {
```
Comment (aledsage) [inline locations/jclouds/src/test/java/brooklyn/entity/rebind/persister/jclouds/BlobStoreExpiryTest.java]: This test assumes the endpoint will be Swift, but nothing in the class explicitly says that the test endpoint must be swift (there is just an "e.g." in some example brooklyn.properties). Would be worth being explicit about that requirement.

Note this was a problem with the code before @neykov improved it, so I'm not too bothered if it's included in this PR or another.

**Comment by neykov** [comment]
Resolved merge conflicts and addressed comment.

**Comment by aledsage** [comment]
Looks good; merging.

**Pull request merged**